### PR TITLE
feat: add rootClassName

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/34318
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      add `rootClassName`     |
| 🇨🇳 Chinese |    添加 `rootClassName`       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: src/Menu.tsx [hunk 1]
```diff
@@ -33,15 +33,18 @@ import { useImperativeHandle } from 'react';
...
  * ## Remove
- * - openTransitionName
- * - openAnimation
- * - onDestroy
- * - siderCollapsed: Seems antd do not use this prop (Need test in antd)
- * - collapsedWidth: Seems this logic should be handle by antd Layout.Sider
+ *
+ * - OpenTransitionName
+ * - OpenAnimation
+ * - OnDestroy
```
Comment (zombieJ) [inline src/Menu.tsx:45]: 备注里的是方法，不用转换大小写

File: tests/__snapshots__/SubMenu.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          class="rc-menu-submenu-arrow"
+        />
+      </div>
+      <div>
+        <div
+          class="rc-menu-submenu rc-menu-submenu-popup custom-className"
+          style="opacity: 0; pointer-events: none;"
+        >
+          <ul
+            class="rc-menu rc-menu-sub rc-menu-vertical custom-className"
```
Comment (zombieJ) [inline tests/__snapshots__/SubMenu.spec.js.snap:90]: rootClassName 总是出现在最外层 dom 上，这边出现两个是不对的。应该就在最外面的 div 上面。

File: src/Menu.tsx (hunk 1 above)
Comment (heiyu4585) [reply src/Menu.tsx:45]: prettier自动转化的,也不转么

File: tests/__snapshots__/SubMenu.spec.js.snap (hunk 2 above)
Comment (heiyu4585) [reply tests/__snapshots__/SubMenu.spec.js.snap]: ready

File: src/Menu.tsx (hunk 1 above)
Comment (heiyu4585) [reply src/Menu.tsx]: ready

**Comment by zombieJ** [comment]
@MadCcc pls help to release related minor version.

**Comment by MadCcc** [comment]
v9.5.4 published

**Pull request merged**